Db updates

irfamustafa opened the pull request:
Updated the DB Schema and corresponding insert statements by removing the type table and just placing a category (i.e. type) column into modules table

irfamustafa committed:
Add files via upload

Added in scores into one of the tables (i.e. Employees_Take_Modules)

irfamustafa committed:
Add files via upload

Added in the score value into one of the insert statements to match the change in the DB Schema

irfamustafa committed:
Add files via upload

Added in score value into one of the insert statements to match the change in the DB Schema

irfamustafa committed:
Add files via upload

Updates in the insert statement: removed the type(i.e. category) insert statements since type(i.e. category) is now just a column in the modules table.

irfamustafa committed:
Add files via upload

Schema Updates: removed the type (i.e category) table and made it a column in the modules table; linked Stream table to modules instead

irfamustafa committed:
Add files via upload

Updated ER Diagram: to match the updated DB Schema with the type table removed

kabollinger03 commented on DBInsertData.sql:
```diff
@@ -8,14 +8,14 @@ Insert INTO Class VALUES ('JV123', 'FSD123', 'IN5');
 
-Insert INTO Type VALUES('Java', 'FSD123');
-Insert INTO Type VALUES('DB', 'DB343');
+--Insert INTO Type VALUES('Java', 'FSD123');
```
These don't need to be commented out, because the script is going to be run by someone who has never run it before.

Lyx-22 commented:
I think there should be a isValidated column in Users to make sure the the email address (username) is correct. We should not store passwords inside the DB for security reasons and use a hashed value instead. I am not sure if should have an extra column in Users for ConfirmPassword (When I learned Ruby on Rails, the table would have this extra column). This extra Password field would be used when changing passwords.

kabollinger03 commented on DBSchemaTables.sql:
```diff
@@ -71,12 +71,13 @@ CREATE TABLE Student_Performance.Type (
   module_id NUMBER(10) NOT NULL,
   module_name VARCHAR2(45) NULL,
-  module_type VARCHAR2(30) NOT NULL,
+  category VARCHAR2(30) NULL,
```
Category should be not null

kabollinger03 commented:
users need username to be varcher2(100)
users password should probably be varchar2(30)
class doesn't need a column for user_id
employees name should go up to 90 or 100 to mirror change in users.username
employees email should go up to 100, also to mirror this change 
courses may need a big VARCHAR for description, but that is totally optional for now
Why do we need the table Class_has_Modules? For one, it is stream_has_modules and class_has_stream. For two, class has a module_id column, and modules has a stream_id column,

irfamustafa committed:
Add files via upload

Removed the class_has_modules table so removed those insert statements

irfamustafa committed:
Add files via upload

Removed the Class_Has_Modules table since class connects to stream and stream connects to modules. Also fixed some of the spaces allocated to certain variables such as changing user_id varchar2(30) to user_id varchar2(100).

irfamustafa committed:
Add files via upload

Please look at this image for the latest updates to the ER Diagram. The changes are typed out in the text boxes.

kabollinger03 commented on DBSchemaTables.sql:
```diff
@@ -25,7 +26,7 @@ CREATE TABLE Student_Performance.Stream (
   class_id VARCHAR2(30) NOT NULL,
   stream_id VARCHAR2(40) NOT NULL,
-  user_id VARCHAR2(30) NOT NULL,
+  user_id VARCHAR2(100) NOT NULL,
```
Doesn't /Need/ a user_id but there is no harm in keeping it.

kabollinger03 reviewed (approved):
Minor comments, not hurting anything really.

The type table (type-stream I think is the purpose) can go away but it isn't hurting anything. Streams don't have types, modules do. And module type is now a column.

Courses could possibly have a description column, but I would not worry too much about that.

kjones2296 commented:
Kate and Kevin approved changes to database
